tests/xmltester/XMLTester.cpp

```diff
@@ -843,31 +841,84 @@ XMLTester::areaDelta(const geom::Geometry* a, const geom::Geometry* b, std::stri
         ss << *geomSD;
         ss << std::endl;
     }
-
     return diffScore;
 }
 
+void
+Test::checkResult( const std::unique_ptr<Geometry>& result ) 
```

Comment:
Taking a reference to a `unique_ptr` here (and elsewhere) strikes me as odd...why not `const Geometry& result` ?

Reply:
great, will try that out.

Reply:
One reason is that this requires the caller to use `*gUniquePtr`, which always strikes me as ugly (and slightly magical)

Reply:
Dereferencing a pointer with `*` is pretty standard C++ usage, though.

Reply:
After some reading ([here](https://stackoverflow.com/questions/30905487/how-can-i-pass-stdunique-ptr-into-a-function) and [here](https://github.com/CppCon/CppCon2014/blob/master/Presentations/Back%20to%20the%20Basics!%20Essentials%20of%20Modern%20C%2B%2B%20Style/Back%20to%20the%20Basics!%20Essentials%20of%20Modern%20C%2B%2B%20Style%20-%20Herb%20Sutter%20-%20CppCon%202014.pdf)) I'm convinced.  The ownership of the parameter is an external concern which should not be exposed inside the functions.